### PR TITLE
add Metadata  for BOM (#6)

### PR DESCRIPTION
**Pull request by Bertk**
related to issue #6 

- option to import Bom.Metadata using XML template
- add or update Bom.Metadata.Tools information

### CODE REVIEW

File: CycloneDX/Program.cs [hunk 1]
```diff
@@ -257,5 +284,48 @@ public static async Task<int> Main(string[] args)
...
+            catch (IOException ex)
+            {
+                Console.Error.WriteLine($"Could not read Metadata file.");
+                Console.WriteLine(ex.Message);
+            }
+            return bom;
+        }
+        static public void AddMetadataTool(Bom bom)
+        {
+            string toolname = "CycloneDX module for .NET";
```
Comment (stevespringett) [inline CycloneDX/Program.cs:303]: Why does this default to "CycloneDX module for .NET"? The tools cannot simply make thing up.

Comment (Bertk) [reply CycloneDX/Program.cs:303]: I used the title from [README.md](https://github.com/CycloneDX/cyclonedx-dotnet/blob/master/README.md#cyclonedx-module-for-net) and this describes the tool which was used to create the BOM. The [CycloneDX Node.js Module](https://www.npmjs.com/package/@cyclonedx/bom) creates a similar **Tool** entry. 
My preferred choice would be the details from **CycloneDX.exe** which uses product name "_CycloneDX_" but this is not helpful.

image

Comment (Bertk) [reply CycloneDX/Program.cs:303]: @stevespringett Do you have any proposals or recommendation for this PR?

Comment (Bertk) [reply CycloneDX/Program.cs:303]: The metadata tools shows the creator tool for the BOM (This is similar to npm tool)

**Comment by coderpatros** [comment]
Thanks for the PR @Bertk

**Pull request merged**